src/core/tsi/transport_security_grpc.h

```diff
@@ -69,6 +80,9 @@ struct tsi_zero_copy_grpc_protector_vtable {
   tsi_result (*unprotect)(tsi_zero_copy_grpc_protector* self,
                           grpc_slice_buffer* protected_slices,
                           grpc_slice_buffer* unprotected_slices);
+  tsi_result (*unprotect_and_get_min_progress_size)(
+      tsi_zero_copy_grpc_protector* self, grpc_slice_buffer* protected_slices,
+      grpc_slice_buffer* unprotected_slices, int* min_progress_size);
```

Comment:
This looks good to me overall, but have we considered the possibility to add `int* min_progress_size` directly to the `unprotect` function signature? Because right now the two functions are almost the same, except for the last parameter.
How difficult it is if we change all the internal call-sites of TSI to fill in `min_progress_size`? I think doing that might require some extra work, but is generally beneficial to the code cleanliness in the long run.

Reply:
Yeah Craig and I talked about this. The cherrypick will unfortunately be large if we change  the unprotect method directly and we have to get LGTMs from all the teams/users for landing or rolling back this change. This way, we can incrementally switch each user one by one to use the new method and then after everyone is migrated we can delete the old method and rename this in one go.

Reply:
OK, sounds good. I am fine with that approach as well.
It would be good if we could have a TODO for that :)

Reply:
Thanks. Added a TODO

Reply:
I agree with Zhen on this.  I would really prefer to see this parameter added to the existing method rather than cluttering the API with new method variants this way.

It's not clear to me how it helps the import to add a new method anyway.  Either way, you will need to change all existing implementations, whether that's to add a new method or to add a parameter to the existing method.

Reply:
Hey Mark, we're not talking a few callsites here... it's a fairly large change.
The plan here is to introduce the new function, migrate incrementally, and then delete the old function - effectively eliminating the old method (this was mentioned up-thread).

Not doing so for this one would be too risky - it's going to be an LSC internally to get it done as it is.

Reply:
This PR is creating a new method only for the zero-copy frame protector API, which has no callers outside of gRPC.  In codesearch, I see only one reference to `tsi_zero_copy_grpc_protector_unprotect()` internally, and it's the one that the cherry-pick import CL for this PR is already touching.

Reply:
I'll remove this method and do the cherrypick for this one. A similar change has to be made for the non_zero_copy based protector as well that cherrypick would be much larger.

Reply:
Sounds good, thanks!

Another option to consider is adding the parameter but making it an optional parameter, so that there's no need to modify any caller.

Reply:
Craig doesn't like optional parameters I think :) Its probably better to force every caller to use it. I'll make the change. I got confused before with regards to which cherrypick would be large (zero copy vs non-zero-copy). There don't appear to be a lot of callers for zero copy protect/unprotect

Reply:
Renamed the methods now.